MLP_Model/Hyperparameter_Optimization_model.ipynb

```diff
@@ -0,0 +1,3576 @@
+            "ticks": ""
+           }
+          },
+          "scene": {
+           "xaxis": {
+            "backgroundcolor": "#E5ECF6",
+            "gridcolor": "white",
+            "gridwidth": 2,
+            "linecolor": "white",
+            "showbackground": true,
+            "ticks": "",
+            "zerolinecolor": "white"
+           },
+           "yaxis": {
+            "backgroundcolor": "#E5ECF6",
+            "gridcolor": "white",
+            "gridwidth": 2,
+            "linecolor": "white",
+            "showbackground": true,
+            "ticks": "",
+            "zerolinecolor": "white"
+           },
+           "zaxis": {
+            "backgroundcolor": "#E5ECF6",
+            "gridcolor": "white",
+            "gridwidth": 2,
+            "linecolor": "white",
+            "showbackground": true,
+            "ticks": "",
+            "zerolinecolor": "white"
+           }
+          },
+          "shapedefaults": {
+           "line": {
+            "color": "#2a3f5f"
+           }
+          },
+          "ternary": {
+           "aaxis": {
+            "gridcolor": "white",
+            "linecolor": "white",
+            "ticks": ""
+           },
+           "baxis": {
+            "gridcolor": "white",
+            "linecolor": "white",
+            "ticks": ""
+           },
+           "bgcolor": "#E5ECF6",
+           "caxis": {
+            "gridcolor": "white",
+            "linecolor": "white",
+            "ticks": ""
+           }
+          },
+          "title": {
+           "x": 0.05
+          },
+          "xaxis": {
+           "automargin": true,
+           "gridcolor": "white",
+           "linecolor": "white",
+           "ticks": "",
+           "title": {
+            "standoff": 15
+           },
+           "zerolinecolor": "white",
+           "zerolinewidth": 2
+          },
+          "yaxis": {
+           "automargin": true,
+           "gridcolor": "white",
+           "linecolor": "white",
+           "ticks": "",
+           "title": {
+            "standoff": 15
+           },
+           "zerolinecolor": "white",
+           "zerolinewidth": 2
+          }
+         }
+        },
+        "title": {
+         "text": "Optimization History Plot"
+        },
+        "xaxis": {
+         "title": {
+          "text": "Trial"
+         }
+        },
+        "yaxis": {
+         "title": {
+          "text": "Objective Value"
+         }
+        }
+       }
+      }
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "fig = optuna.visualization.plot_optimization_history(study)\n",
+    "graph_path = f\"./figures/{params.MODEL_TYPE}/plot_optimization_history_graph\"\n",
+    "fig.write_html(Path(f\"{graph_path}.html\"))\n",
+    "fig.write_image(Path(f\"{graph_path}.png\"))\n",
+    "fig.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "92103de1",
+   "metadata": {},
+   "outputs": [
```

Comment:
Minor suggestion: I noticed the output for this and other cells aren't visible within GitHub or ReviewNB. Recognizing this may be a limitation of how those platforms render this content, are there other ways of ensuring an executed notebook with these types of visualizations are stored within source control somehow? You've done your due diligence in storing multiple kinds of files for these (nice work!), so this is mostly wondering about strengthening presentational / contextual information that comes along with reading the notebook file and the diagrams together.

Reply:
I think these outputs are in the output of images of graphs. I am not sure though let me check